library/Fission/Web.hs

```diff
@@ -37,47 +33,27 @@ import qualified Fission.Web.User    as User
 type API = Web.Swagger.API :<|> Web.API
 
 app ::
-  ( MonadLocalIPFS        (RIO cfg)
-  , MonadRemoteIPFS       (RIO cfg)
-  , MonadLinkedIPFS       (RIO cfg)
-  , MonadTime             (RIO cfg)
-  , MonadDB               (RIO cfg)
-  , MonadDNSLink          (RIO cfg)
-  , MonadLogger           (RIO cfg)
-  , MonadReflectiveServer (RIO cfg)
-  , MonadReader                cfg m
-  , MonadHerokuAddOn               m
-  , MonadReflectiveServer          m
+  ( MonadDB               m
+  , MonadTime             m
+  , MonadLogger           m
+  , MonadDNSLink          m
+  , MonadLocalIPFS        m
+  , MonadRemoteIPFS       m
+  , MonadLinkedIPFS       m
+  , MonadReflectiveServer m
   )
-  => m Application
-app = do
-  cfg     <- ask
-  auth    <- mkAuth
-  appHost <- getHost
-
+  => (forall a . m a -> Handler a)
```

Comment:
Question: What does this line...um...mean? 😅

Reply:
@bechurch 

Good question! This is not a common thing, so don't worry about it too much. I did this to clean up the constraints. It's not _required_, but is much cleaner this way / a nice to have.

# Short Answer

We're defining a locally scoped type variable (inside the parens).

# Long Answer

Every Haskell function has an implicit `forall` at the beginning of the type signature. For example:

```haskell
myLogInfo :: (Monad m, Show a) => a -> m a

-- is actually
myLogInfo :: forall m a . (Monad m, Show b) => a -> m a
```

This is called "universal quantification", just like in logic. "For all `a`s, the following relationships holds true."

We can also nest these inside the type of passed functions. For example:

```haskell
```

Reply:
Ahh gotcha! Thats what was tripping me up I was already reading types like boolean algebra and wondering if we didn't specify it before why were we now.